File: modules/agent/src/models/UserPreferences.py
from sqlalchemy import Column, Text, DateTime, BigInteger, ForeignKey, func
from sqlalchemy.orm import relationship
from models import Base


class UserPreferences(Base):
    """UserPreferences SQLAlchemy model (theme and locale settings for individual user)."""

    __tablename__ = "User_Preferences"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    user_id = Column(BigInteger, ForeignKey("User.id", ondelete="CASCADE"), nullable=False, unique=True, index=True)
    theme = Column(Text, nullable=True)
    timezone = Column(Text, nullable=True, default="America/New_York")
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    user = relationship("User", back_populates="preferences", uselist=False, foreign_keys=[user_id])

File: modules/agent/internal/models/UserPreferences.go
package models

import "time"

type UserPreferences struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID    int64     `gorm:"uniqueIndex;not null" json:"user_id"`
	Theme     *string   `json:"theme"` // null = inherit from tenant
	Timezone  *string   `json:"timezone"`
	CreatedAt time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (UserPreferences) TableName() string {
	return "User_Preferences"
}
